coil-base/src/main/java/coil/memory/RequestService.kt

```diff
@@ -145,10 +147,10 @@ internal class RequestService {
         return transformations.isEmpty() || Transformation.VALID_CONFIGS.contains(bitmapConfig)
     }
 
-    private fun Request.isConfigValidForHardware(size: Size): Boolean {
+    private fun Request.isConfigValidForHardware(size: Size, target: Target?): Boolean {
         if (SDK_INT < O) return true
         if (bitmapConfig != Bitmap.Config.HARDWARE) return true
-        return allowHardware && hardwareBitmapService.allowHardware(size)
+        return allowHardware && hardwareBitmapService.allowHardware(size) && (target !is ViewTarget<*> || target.view.isHardwareAccelerated)
```

Comment:
We should check `hardwareBitmapService.allowHardware(size)` last since it can be expensive. Also let's add this target check on a separate line.